src/Kiota.Builder/CodeDOM/CodeClass.cs

```diff
@@ -65,13 +65,49 @@ public bool IsErrorDefinition
     }
     public override IEnumerable<CodeProperty> AddProperty(params CodeProperty[] properties)
     {
-        var result = base.AddProperty(properties);
-        foreach (var addedPropertyTuple in result.Select(x => new Tuple<CodeProperty, CodeProperty?>(x, StartBlock.GetOriginalPropertyDefinedFromBaseType(x.Name)))
-                                        .Where(static x => x.Item2 != null))
-            addedPropertyTuple.Item1.OriginalPropertyFromBaseType = addedPropertyTuple.Item2!;
+        if (properties == null || properties.Any(static x => x == null))
+            throw new ArgumentNullException(nameof(properties));
+        if (!properties.Any())
+            throw new ArgumentOutOfRangeException(nameof(properties));
 
+        var result = new CodeProperty[properties.Length];
+
+        for (var i = 0; i < properties.Length; i++)
+        {
+            var property = properties[i];
+            if (property.IsOfKind(CodePropertyKind.Custom, CodePropertyKind.QueryParameter))
+            {
```

Comment:
this could be simplified by foreach and where

Reply:
Are you sure? This method needs to return an IEnumeration with the properties added. The properties need to be added one by one and only some need to the be modified. The index is used to store the resulting properties in the result array. Maybe my skills with the dotnet API are lacking here. I took `CodeBlock.AddRange` as example.

Reply:
right, not for the where.
But you could use a select on the properties, the body of which would be your for body, and return the toarray value instead of doing what you're doing right now.

Reply:
Yes, this actually is very similar to how Streams work in Java. I've managed to rewrite the code to use Select.